src/main/PurchaseList.java

```diff
@@ -0,0 +1,97 @@
+package main;
+
+import main.exceptions.CsvLineException;
+import main.exceptions.NoArgumentException;
+import main.exceptions.OutOfBoundArgumentException;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.Scanner;
+
+
+public class PurchaseList {
+    private final ArrayList<Purchase> purchases;
+    private final Comparator<Purchase> comparator;
+
+    public PurchaseList(String path, Comparator<Purchase> comparator) throws CsvLineException {
+        this(new File(path), comparator);
+    }
+
+    public PurchaseList(Comparator<Purchase> comparator) throws CsvLineException {
+        this("", comparator);
+    }
+
+    public PurchaseList(File fileName, Comparator<Purchase> comparator) throws CsvLineException {
+        this.purchases = new ArrayList<>(0);
+        this.comparator = comparator;
+        try (Scanner scanner = new Scanner(new FileReader(fileName))){
+            while(scanner.hasNextLine()){
+                Purchase purchase = PurchasesFactory.createPurchase(scanner.nextLine());
+                purchases.add(purchase);
+            }
+            if (purchases.size() == 0) {
+                throw new NoArgumentException("File is empty");
+            }
+        } catch (FileNotFoundException e) {
+            //e.printStackTrace();
+            throw new CsvLineException("file not found", e);
+        } catch (CsvLineException e) {
+            System.err.println(e);
+        }
+    }
+
+    public void subList(int from, int to) throws IllegalArgumentException{
```

Comment:
```suggestion
    public int remove(int from, int to) {
```
return the number of removed elements; 
start with
```java
if(from >= to) {
    return 0;
}
```